detection_rules/rule.py

```diff
@@ -902,6 +934,7 @@ def post_validation(self, value: dict, **kwargs):
 
         data.validate_query(metadata)
         data.data_validator.validate_note()
+        data.validation(metadata) if  hasattr(data, 'validation') else None
```

Comment:
In the `NewTermsRuleData` class is a `validation` method that is unique to that rule type but does further validation outside of Marshmallow. The logic is, build the rule according to the defined dataclass schema and then call this method for further validation. Within a few rule types is the method `validate_query` which is focused on the query of the rule itself. Rather than rely on this, which can be misleading, if a `validation` method exists under any rule type it will be called making this agnostic to other rule types that may need further validation for the rule type itself, outside of the query and Marshmallow.